llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -1966,6 +1987,15 @@ getDependenceDistanceStrideAndSize(
       BPtr->getType()->getPointerAddressSpace())
     return MemoryDepChecker::Dependence::Unknown;
 
+  // Ignore Histogram count updates as they are handled by the Intrinsic. This
+  // happens when the same pointer is first used to read from and then is used
+  // to write to.
+  if (!AIsWrite && BIsWrite && APtr == BPtr && HistogramPtrs.contains(APtr)) {
+    LLVM_DEBUG(dbgs() << "LAA: Histogram: Update is safely ignored. Pointer: "
```

Comment:
I removed most of the debug printing (keeping the main 'found' message), but added a histogram dependency type so that we can test the analysis part independently of any transformation.

I also deferred some of the checks to the cost modelling part of transformation -- the element types supported depend on the target, and we now have the ability to scalarize if needed. The base cost function will need to be updated at some point to reflect that.

Reply:
> removed most of the debug printing  ...
> added a histogram dependency type so that we can test the analysis part independently of any transformation.

Good idea, thanks.